apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -33,10 +33,22 @@ type OpenTelemetryCollectorSpec struct {
 	// +optional
 	Args map[string]string `json:"args,omitempty"`
 
+	// Autoscale turns on/off the autoscale feature. By default, it's enabled if the Replicas field is not set.
+	// +optional
+	Autoscale *bool `json:"autoscale,omitempty"`
+
 	// Replicas is the number of pod instances for the underlying OpenTelemetry Collector
 	// +optional
 	Replicas *int32 `json:"replicas,omitempty"`
 
+	// MinReplicas sets a lower bound to the autoscaling feature.
+	// +optional
+	MinReplicas *int32 `json:"minReplicas,omitempty"`
+
+	// MaxReplicas sets an upper bound to the autoscaling feature. When autoscaling is enabled and no value is provided, a default value is used.
+	// +optional
+	MaxReplicas *int32 `json:"maxReplicas,omitempty"`
```

Comment:
I have a proposal to make the config/CR simpler. 

Could we add only `MaxReplicas` field in this PR to support autostacling? 

If the max replicas is set the operator would create HPA. In the validating webhook it would check if maxReplicas is higher than replicas.

Reply:
It seems that `MaxReplicas` can be nil (e.g. for infinite scaling) in this case we need `Autoscale` flag to enable the HPA.

Reply:
Probably we can do It (remove autoscale), https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v1/#HorizontalPodAutoscalerSpec
because maxReplicas is required:

maxReplicas (int32), required
upper limit for the number of pods that can be set by the autoscaler; cannot be smaller than MinReplicas.